core/src/main/java/com/datastrato/gravitino/storage/relational/service/CatalogMetaService.java

```diff
@@ -0,0 +1,203 @@
+      }
+      SessionUtils.doWithCommit(
+          CatalogMetaMapper.class,
+          mapper -> {
+            CatalogPO po = POConverters.initializeCatalogPOWithVersion(catalogEntity, metalakeId);
+            if (overwrite) {
+              mapper.insertCatalogMetaOnDuplicateKeyUpdate(po);
+            } else {
+              mapper.insertCatalogMeta(po);
+            }
+          });
+    } catch (RuntimeException re) {
+      if (re.getCause() != null
+          && re.getCause().getCause() != null
+          && re.getCause().getCause() instanceof SQLIntegrityConstraintViolationException) {
+        // TODO We should make more fine-grained exception judgments
+        // Usually throwing `SQLIntegrityConstraintViolationException` means that
+        // SQL violates the constraints of `primary key` and `unique key`.
+        // We simply think that the entity already exists at this time.
+        throw new EntityAlreadyExistsException(
+            String.format(
+                "Catalog entity: %s already exists", catalogEntity.nameIdentifier().name()));
+      }
+      throw re;
+    }
+  }
+
+  public <E extends Entity & HasIdentifier> CatalogEntity updateCatalog(
+      NameIdentifier identifier, Function<E, E> updater) throws IOException {
+    checkCatalogNamespaceWithIdentifier(identifier);
+    String metalakeName = identifier.namespace().level(0);
+    String catalogName = identifier.name();
+    Long metalakeId =
+        SessionUtils.getWithoutCommit(
+            MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeIdMetaByName(metalakeName));
+    if (metalakeId == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_AN_ENTITY_MESSAGE, identifier.namespace().toString());
+    }
+
+    CatalogPO oldCatalogPO =
+        SessionUtils.getWithoutCommit(
+            CatalogMetaMapper.class,
+            mapper -> mapper.selectCatalogMetaByMetalakeIdAndName(metalakeId, catalogName));
+    if (oldCatalogPO == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_AN_ENTITY_MESSAGE, identifier.toString());
+    }
+
+    CatalogEntity oldCatalogEntity =
+        POConverters.fromCatalogPO(oldCatalogPO, identifier.namespace());
+    CatalogEntity newEntity = (CatalogEntity) updater.apply((E) oldCatalogEntity);
+    Preconditions.checkArgument(
+        Objects.equals(oldCatalogEntity.id(), newEntity.id()),
+        "The updated catalog entity id: %s should be same with the catalog entity id before: %s",
+        newEntity.id(),
+        oldCatalogEntity.id());
+
+    Integer updateResult =
+        SessionUtils.doWithCommitAndFetchResult(
+            CatalogMetaMapper.class,
+            mapper ->
+                mapper.updateCatalogMeta(
+                    POConverters.updateCatalogPOWithVersion(oldCatalogPO, newEntity, metalakeId),
+                    oldCatalogPO));
+
+    if (updateResult > 0) {
```

Comment:
If the `newEntity` is equal to the old one, the update result is still greater than 0?

Reply:
Yes, I think the update result will be greater than 0. Is it necessary to verify that different attributes must exist before it can be updated?

Reply:
> Is it necessary to verify that different attributes must exist before it can be updated?

I don't think so if the update operation is lightweight.